### PR TITLE
fix: Move generated theme files to frontend/generated

### LINKED ISSUE #11180: theme-name.generated.js should be created to /frontend/generated/ with Flow 2.6+
### Description of the bug / feature
 - My objective is create a custom theme in an separate maven project, and join by a dependency in the pom.
### Minimal reproducible example
 - For that purpose I followed these steps...
   (1) Go to https://start.vaadin.com/ and create new initializr project with and example
   (2) Create new maven project in my own environment and following vaadin doc specification to create a package...
https://vaadin.com/docs/v14/flow/styling/custom-theme-packaging . Theme project is attached here in the issue.
   (3) Maven clean-install in the theme project, and clean-install in the initializr app created.
   (4) Run everything with Spring boot
### Expected behavior
 - With the initializr app, everything was perfect and theme was loading and showing ok.
### Actual behavior
 - In my own project, I'm using as well this maven project theme, but I'm getting following error...
_TypeError: can't access property "appendChild", e.shadowRoot is null_

image

could be is something related with Lumo javascript, or I'm not loading properly lumo javascript clases?

I have also described the issue here in stackoverflow...

https://stackoverflow.com/questions/67874325/why-my-own-theme-does-not-work-with-vaadin-14?noredirect=1#comment119988225_67874325

### Versions:
    - Vaadin / Flow version: 14.6.2 (boht, my project, and theme project)
    - Java version: 11
    - OS version: Windows 10 profesional
    - Browser version (if applicable): Chrome Version 91.0.4472.77 (Official Build) (64-bit)
    - Application Server (if applicable):
    - IDE (if applicable): Eclipse

I will atach projects here...

[eo-vaadin-alonetheme.zip](https://github.com/vaadin/flow/files/6615961/eo-vaadin-alonetheme.zip)

In my project (I can't upload here, is a company project)... I just add to pom.xml...

`<dependency>
 			<groupId>com.vaadin.themes</groupId>
  			<artifactId>eo-vaadin-alonethemes</artifactId>
    		<version>0.0.1-SNAPSHOT</version>
		</dependency>`

and in the class I changed ... 

_@Theme(Lumo.class)_

to

_@Theme(themeFolder = "eo-vaadin-orange")_

### PR DESCRIPTION
**Pull request by caalador**
Any theme file that is generated is
now generated into `frontend/generated`

Fixes #11180

### CODE REVIEW

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ function generateThemeFile(themeFolder, themeName, themeProperties, productionMo
   }
 
   if (themeProperties.parent) {
-    themeFile += `import {applyTheme as applyBaseTheme} from 'themes/${themeProperties.parent}/${themeProperties.parent}.generated.js';`;
+    themeFile += `import {applyTheme as applyBaseTheme} from 'generated/${themeProperties.parent}.generated.js';`;
```
Comment (pleku) [inline flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js:137]: This is different from `master` branch which has 
```javascript
themeFile += `import {applyTheme as applyBaseTheme} from './theme-${themeProperties.parent}.generated.js';`;
```

There are other changes too that can be seen with `git diff master issues/11180_move_generated_theme_file -- flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js`

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js [hunk 2]
```diff
@@ -23,8 +23,8 @@ const path = require('path');
 const generateThemeFile = require('./theme-generator');
 const {copyStaticAssets, copyThemeResources} = require('./theme-copy');
 
-// matches theme folder name in 'themes/my-theme/my-theme.generated.js'
-const nameRegex = /themes\/(.*)\/\1.generated.js/;
+// matches theme name 'my-theme' in 'generated/my-theme.generated.js'
+const nameRegex = /generated\/(.*).generated.js/;
```
Comment (pleku) [inline flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js:27]: The is a diff here too to `master`, seen with `git diff master issues/11180_move_generated_theme_file -- flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js`

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js [hunk 3]
```diff
@@ -152,7 +152,7 @@ function handleThemes(themeName, themesFolder, options, logger) {
     copyThemeResources(themeFolder, options.projectStaticAssetsOutputFolder, logger);
     const themeFile = generateThemeFile(themeFolder, themeName, themeProperties, !options.devMode);
 
-    fs.writeFileSync(path.resolve(themeFolder, themeName + '.generated.js'), themeFile);
+    fs.writeFileSync(path.resolve(options.frontendGeneratedFolder, themeName + '.generated.js'), themeFile);
```
Comment (pleku) [inline flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js:155]: In `master` the file is written as `(options.frontendGeneratedFolder, 'theme-' + themeName + '.generated.js'), themeFile);`

which one it should be ?

File: .gitignore [hunk 4]
```diff
@@ -47,3 +47,4 @@ flow-tests/**/types.d.ts
 package.json
 package-lock.json
 /flow-tests/**/*.generated.js
+/flow-tests/**/frontend/generated/
```
Comment (pleku) [inline .gitignore:50]: This should be added to the V14 branches on starters too, as probably newer branches have it already due to fusion

**Review by pleku — CHANGES_REQUESTED** [review]

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js (hunk 3 above)
Comment (caalador) [reply flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js:155]: For this it is `[themeName].generated.js` I don't know why master has added a `theme-` to the generated name.
The same is why was `theme-generated.js` not ok and was changed to `theme.js` in master?
The other diffs are also a bit related to that.

Comment (caalador) [reply flow-server/src/main/resources/plugins/application-theme-plugin/theme-handle.js]: Did the renaming to match master

**Review by pleku — APPROVED** [review]

**Comment by pleku** [comment]
Intentionally not picking this to 2.6 to reduce regression risks.

**Review by pleku — APPROVED** [review]

**Pull request merged**